LanguageFeatures/Patterns/patterns_legacy_lib.dart

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @description Defines data used by Patterns tests to check interaction with
+/// legacy
+/// @author sgrekhov22@gmail.com
+
+// @dart=2.9
```

Comment:
We shouldn't do this, because it will not be supported when patterns are part of the language, it's very very short-lived.

Reply:
Actually, I'd be surprised if it will ever be possible to run this: Isn't it already a compile-time error to request unsound-null-safety execution when patterns or records are enabled?

Anyway, it's too short-lived to be useful.

Reply:
I can't run it manually, but via `test.py` this test works. Just to confirm, we are going to delete this test and leave the statement below (and the similar statemwent for switch-expression) untested (as almost not-actual). Correct?

> If the static type of v is an always-exhaustive type, no case matches, and there is no default clause, then throw a runtime error. This can only occur when null or a legacy typed value flows into this switch statement from another library that hasn't migrated to [null safety](https://dart.dev/null-safety). In fully migrated programs, exhaustiveness checking is sound and it isn't possible to reach this runtime error.

Reply:
Deleted

Reply:
Yes, there's no need to test unsound null safety semantics with patterns, they will never be available to the public together (and it doesn't matter that we can have that combination right now, basically by accident).